server/datastore/mysql/hosts.go

```diff
@@ -546,17 +546,31 @@ func (ds *Datastore) CountHosts(ctx context.Context, filter fleet.TeamFilter, op
 	return count, nil
 }
 
-func (ds *Datastore) CleanupIncomingHosts(ctx context.Context, now time.Time) error {
-	sqlStatement := `
+func (ds *Datastore) CleanupIncomingHosts(ctx context.Context, now time.Time) ([]uint, error) {
+	var ids []uint
+	selectIDs := `
+    SELECT
+      id
+    FROM
+      hosts
+    WHERE
+      hostname = '' AND
+      osquery_version = '' AND
+      created_at < (? - INTERVAL 5 MINUTE)`
```

Comment:
I guess this is a little bit paranoid but, the `created_at < (....)` here and the `created_at < (...)` used in the `DELETE` below can potentially return different results.

Reply:
In practice it can't - it uses the same WHERE clause with a fixed timestamp using the `created_at` column as condition, so it cannot create new rows 5+ minutes in the past.

Reply:
Thanks @mna, you are right (seems like I need a little bit more coffee)